server/main.go

```diff
@@ -256,28 +312,40 @@ func sendWorldState(id string) error {
 	player, ok := players[id]
 	playersLock.RUnlock()
 	if !ok {
+		log.Print(errors.New(fmt.Sprintf("Client disconnected: (cant find ID: %s)", id), nil))
+		playersLock.Lock()
+		delete(players, id)
+		playersLock.Unlock()
+		queueUpdate(&update{
+			notifyPlayerDisconnected: &notifyPlayerDisconnected{
+				id: id,
+			},
+		})
```

Comment:
i don't understand the purpose of this block. if `ok == false`, then `delete(players, id)` is noop

also, unless there is a bug somewhere in the program, `ok` should never `== true` if the player is still connected. if a player disconnects, we should handle the disconnect when it occurs (queue the update notify player disconnected *then*, not wait til this loop to catch it)

i'd be in favor of removing this whole block